### ISSUE TITLE
AGRC Base Map Selector Widget

### ISSUE THREAD

**Issue by stdavis** [agrc/wfrc-wab-widgets#19]
From @bertgranberg:
>i would _love_ to have that, if you'd got time...with the full set of choices add points, landown, overlay
>especially if we could place it under the +/- map controls

**Comment by msilski** [comment]
On the same thread, our Basemap Gallery widget that contains a mix of ESRI and AGRC basemaps is not showing the AGRC basemaps (Utah Hybrid Basemap and Utah Lite Basemap). Do you know of a configuration/setting that would allow those basemaps to display?

**Comment by stdavis** [comment]
I just gave Bert a new quad word to try. Let me know if it works...

**Comment by msilski** [comment]
It works!

**Comment by bertgranberg** [comment]
Back to the original comment, even though we have the base maps working with the stock WAB selector, I still would prefer to use the AGRC base map selector widget positioned either on the left side of the map frame. I think the land ownership, address point and aerials with or without overlayed vectors are better. And, I'd like to keep moving away from the screen clogging WAB widgets.

**Comment by stdavis** [comment]
Oh yes. I was thinking that this was a separate issue that adding the AGRC base map selector.

**Comment by stdavis** [comment]
I just pushed a new release with a widget that wraps the AGRC base map selector called LayerSelector. It's close but I'm not sure that it's 100% ready yet. I've seen some instances were it loads layers in the wrong order so that the overlays are under the base maps. I wonder if this is due to the fact that it's working on a AGOL web map. I'll investigate further next week. I wanted to get you something to look at before I went out of town.

**Comment by msilski** [comment]
I will test it locally and show Bert. Thanks for your work on this.

**Comment by stdavis** [comment]
@msilski Any feedback from your testing of this widget?

**Comment by msilski** [comment]
I tested it a couple weeks ago, but I don't think I correctly followed the instructions in the Widget Documentation. I will take another look at it next week.

**Comment by msilski** [comment]
Just to make sure, when I configure the Layer Selector widget, do I need to follow the same configuration steps in the URLParams widget? Ex. "Additional steps to add as an on-screen widget to WAB Developer Edition" and "Additional steps to add as an on-screen widget to an existing WAB Application?"

Also, I can't find the quad word I used to resolve the other basemap issue back in October. Do you mind sending that over again? Thanks.

**Comment by stdavis** [comment]
Waiting on a new web map from @msilski...

**Comment by stdavis** [comment]
Oops. Wrong issue. Disregard...

**Comment by stdavis** [comment]
Waiting on results of testing by Matthew.

**Comment by msilski** [comment]
I was able to get the widget in our test application with our production ArcGIS Online quad word in the config file, but the basemaps are not displaying. I followed the configuration steps under URL Params for the Layer Selector (both the WAB Developer steps and existing WAB app steps). And I set the ArcGIS Online basemap transparency to 100%. Here's what I have:

![layer selector 1](https://user-images.githubusercontent.com/1699870/49834692-49181880-fd5a-11e8-9389-edf81a2ee417.png)

Is there something I need to do to configure the widget to display a certain basemap? Would you be available for a screenshare sometime this week to work through it?

**Comment by stdavis** [comment]
I'm guessing that you have an incorrect quad word. What domain are you using to test with?

**Comment by msilski** [comment]
My testing has been done on my local machine with the WAB Developer edition, using the chance-source-liquid-nelson quad word. Would the quad word work if I tested it in production (wfrc.org/rtp-2019-phasing)?

**Comment by stdavis** [comment]
`chance-source-liquid-nelson` will only work if the app is hosted on wfrc.maps.arcgis.com or wfrc.org. For testing from your local machine, you need to use your wide open quad word (please don't share this on a public site like GitHub). Your wide open quad word starts with `slogan`.

**Comment by msilski** [comment]
It works! I see that the first item in the config under baseLayers is the default, so I reordered the basemaps to Lite, Imagery, Terrain. For now, I will keep it in testing and have Bert take a look at it. And when it goes live, I will change the config to use the ArcGIS Online quad word. I will also remove the Basemaps widget from the app. Thanks for your help!

**Comment by stdavis** [comment]
The black and white aerials will be in the next release.
